prov/efa/src/rdm/rxr_ep.c

```diff
@@ -934,14 +896,36 @@ static int rxr_ep_ctrl(struct fid *fid, int command, void *arg)
 {
 	ssize_t ret;
 	struct rxr_ep *ep;
+	struct ibv_qp_init_attr_ex attr_ex = { 0 };
 	char shm_ep_name[EFA_SHM_NAME_MAX], ep_addr_str[OFI_ADDRSTRLEN];
 	size_t shm_ep_name_len, ep_addr_strlen;
 
 	switch (command) {
 	case FI_ENABLE:
 		ep = container_of(fid, struct rxr_ep, base_ep.util_ep.ep_fid.fid);
 
-		ret = fi_enable(ep->rdm_ep);
+		assert(EFA_EP_TYPE_IS_RDM(ep->base_ep.domain->info));
+		attr_ex.cap.max_send_wr = ep->base_ep.info->tx_attr->size;
+		attr_ex.cap.max_send_sge = ep->base_ep.info->tx_attr->iov_limit;
+		attr_ex.send_cq = ibv_cq_ex_to_cq(ep->ibv_cq_ex);
+
+		attr_ex.cap.max_recv_wr = ep->base_ep.info->rx_attr->size;
+		attr_ex.cap.max_recv_sge = ep->base_ep.info->rx_attr->iov_limit;
+		attr_ex.recv_cq = ibv_cq_ex_to_cq(ep->ibv_cq_ex);
+
+		attr_ex.cap.max_inline_data = ep->base_ep.domain->device->efa_attr.inline_buf_size;
+
+		attr_ex.qp_type = IBV_QPT_DRIVER;
+		attr_ex.comp_mask = IBV_QP_INIT_ATTR_PD | IBV_QP_INIT_ATTR_SEND_OPS_FLAGS;
+		attr_ex.send_ops_flags = IBV_QP_EX_WITH_SEND;
+		if (efa_domain_support_rdma_read(ep->base_ep.domain))
+			attr_ex.send_ops_flags |= IBV_QP_EX_WITH_RDMA_READ;
+		attr_ex.pd = rxr_ep_domain(ep)->ibv_pd;
+
+		attr_ex.qp_context = ep;
+		attr_ex.sq_sig_all = 1;
```

Comment:
Is it possible to clean this up a bit? e.g.:

```c
		attr_ex = {
			.cap = {
				.max_send_wr     = ep->base_ep.info->tx_attr->size,
				.max_send_sge    = ep->base_ep.info->tx_attr->iov_limit,
				.max_recv_wr     = ep->base_ep.info->rx_attr->size,
				.max_recv_sge    = ep->base_ep.info->rx_attr->iov_limit,
				.max_inline_data = ep->base_ep.domain->device->efa_attr.inline_buf_size
			},
			.send_cq        = ibv_cq_ex_to_cq(ep->ibv_cq_ex),
			.recv_cq        = ibv_cq_ex_to_cq(ep->ibv_cq_ex),
			.qp_type        = IBV_QPT_DRIVER,
			.comp_mask      = IBV_QP_INIT_ATTR_PD | IBV_QP_INIT_ATTR_SEND_OPS_FLAGS,
			.send_ops_flags = efa_domain_support_rdma_read(ep->base_ep.domain)
			                ? IBV_QP_EX_WITH_SEND | IBV_QP_EX_WITH_RDMA_READ
			                : IBV_QP_EX_WITH_SEND,
			.pd             = rxr_ep_domain(ep)->ibv_pd,
			.qp_context     = ep,
			.sq_sig_all     = 1,
		};
```

Reply:
i can fix this when I'm back in 2 weeks

Reply:
@darrylabbate Can we merge now and do the refactor later?